Enhanced Randomness through Seeding

FedericoMontini98 opened the pull request:
Made modifications to the files to infuse seed-based randomness into the Mersenne Twister algorithm, enhancing the random address functionality. A new parameter, ‘seed’, has been introduced in .atp files. In the absence of an explicitly set seed, ATP will autonomously generate an initial random seed using the random_device() function. However, if a seed is specified, it will be relayed to the random_generator class. This ‘seed’ parameter can be incorporated within the “RandomDesc” message. Given that “RandomDesc” is utilized for both Sizes and Addresses, it can be employed for either one, both, or none of them.

FedericoMontini98 committed:
Enhanced Randomness through Seeding

Made modifications to the files to infuse seed-based randomness into the Mersenne Twister algorithm, enhancing the random address functionality. A new parameter, ‘seed’, has been introduced in .atp files.
In the absence of an explicitly set seed, ATP will autonomously generate an initial random seed using the random_device() function. However, if a seed is specified, it will be relayed to the random_generator class.
This ‘seed’ parameter can be incorporated within the “RandomDesc” message. Given that “RandomDesc” is utilized for both Sizes and Addresses, it can be employed for either one, both, or none of them.

matteoandreozzi commented on packet_desc.cc:
```diff
@@ -45,7 +45,7 @@ void PacketDesc::init(const uint64_t parentId,
         waitFor = from.wait_for();
     }
-
+    
```
unecessary whitespace change, please revert

matteoandreozzi commented on packet_desc.cc:
```diff
@@ -55,7 +55,13 @@ void PacketDesc::init(const uint64_t parentId,
+        else{
+            randomSize.init(from.random_size());
+        }
+        printf("* ATP-Engine size generator seed: %ld\n",randomSize.getSeed());
```
we don't use printfs. Please change into LOG macro

matteoandreozzi commented on packet_desc.cc:
```diff
@@ -92,12 +98,20 @@ void PacketDesc::init(const uint64_t parentId,
         } else {
             randomAddress.init(from.random_address());
         }
+        printf("* ATP-Engine address generator seed: %ld\n",randomAddress.getSeed());
```
no printfs, please use LOG

matteoandreozzi commented on proto/tp_config.proto:
```diff
@@ -53,6 +53,10 @@ message RandomDesc {
     optional WeibullDesc weibull_desc = 12;
+
+    // Force seed on Mersenne-Twist 64bit
+    optional uint64 seed = 13 [default = 0];
```
we should not have a default value here.

matteoandreozzi reviewed (changes requested):
Please check inline comments

FedericoMontini98 committed:
PR #12 fixes

- Changed printf into LOG
- Removed default value of seed
- Reverted unnecessary whitespace

FedericoMontini98 commented:
I should have solved the problems
